pandas/core/generic.py

```diff
@@ -3994,11 +4037,10 @@ def _needs_reindex_multi(self, axes, method, level):
     def _reindex_multi(self, axes, copy, fill_value):
         return NotImplemented
 
-    _shared_docs[
-        'reindex_axis'] = ("""Conform input object to new index with optional
-        filling logic, placing NA/NaN in locations having no value in the
-        previous index. A new object is produced unless the new index is
-        equivalent to the current one and copy=False
+    _shared_docs['reindex_axis'] = ("""Conform input object to new index
```

Comment:
I think this should start with a single summary line, like you did for `reindex_like`

Reply:
It seems like 'reindex_axis' will be deprecated in the future so I did not bother with it.